services/form-instances.ts

```diff
@@ -169,6 +169,15 @@ export const updateFormInstance = async (
     comunicaRepo.getFormData(form.formTtl),
   ]);
 
+  if (formData.withHistory && !(await formRepo.hasHistoryItems(instanceId))) {
```

Comment:
you could do this in the promise all above, the form-content service is already pretty slow at times, this could help

Reply:
Hmm I did it here because then it only runs if the form actually keeps history, do you think it would be faster to run it always but in the promise all instead?